clients/factory.go

```diff
@@ -226,14 +223,7 @@ func (f factory) createRestClient(u string, httpClient *http.Client) (*rest.Clie
 		return nil, fmt.Errorf("failed to parse URL %q: %w", u, err)
 	}
 
-	opts := []rest.Option{rest.WithHTTPListener(f.httpListener)}
-	if f.rateLimiterEnabled {
-		opts = append(opts, rest.WithRateLimiter())
-	}
-
-	if f.retryOptions != nil {
-		opts = append(opts, rest.WithRetryOptions(f.retryOptions))
-	}
+	opts := f.restOptions()
 
 	restClient := rest.NewClient(parsedURL, httpClient, opts...)
```

Comment:
I guess we don't even need `opts`
```suggestion
	restClient := rest.NewClient(parsedURL, httpClient, f.restOptions()...)
```